src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java

```diff
@@ -585,6 +585,23 @@ public boolean retainAll(Collection<?> c) {
         throw new NotImplementedException();
     }
 
+    @Override
+    public boolean equals(Object object) {
```

Comment:
locally i deleted this function and tests still pass.  Is this needed?

Reply:
I added this function to mimic the way `equals` work for sets in Java. This does not help fixing the bug, may is a nice to have I think.

Reply:
I have some concerns with adding this. I don't see this being necessary and actually calling [contains](https://github.com/jwplayer/southpaw/blob/master/src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java#L148) for each item is potentially very expensive as it will iterate through the set for each item in the set.